src/buf/into_buf.rs

```diff
@@ -1,6 +1,14 @@
 use super::{Buf};
 
-use std::io;
+#[cfg(feature = "use_alloc")]
+use alloc::string::String;
+#[cfg(feature = "use_alloc")]
+use alloc::vec::Vec;
+
+#[cfg(not(feature = "use_std"))]
+use buf::Cursor;
```

Comment:
@alexcrichton if `buf::Cursor` is defined as a re-export of `std::io::Cursor` when `use_std` is set, is that ok for "additive" features?